enterprise/server/scheduling/scheduler_server/scheduler_server.go

```diff
@@ -205,13 +206,12 @@ func (h *executorHandle) Serve(ctx context.Context) error {
 	}
 	h.groupID = groupID
 
-	var registeredNode *scpb.ExecutionNode
 	removeConnectedExecutor := func() {
-		if registeredNode == nil {
+		if h.registration == nil {
```

Comment:
I think you need a lock around h.registration

Reply:
Done, thanks